### PR TITLE
tuner: Fix missing config.h include

### PR DESCRIPTION
**Pull request by rajachan**
Without this, we were incorrectly including neuron headers instead of CUDA ones when HAVE_CUDA was not defined. Also fail compilation if neither definition was found rather than having a fall-through case.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: include/nccl-headers/error.h [hunk 1]
```diff
@@ -7,7 +7,7 @@
 
 #if HAVE_CUDA
 #include "nccl-headers/nvidia/err.h"
-#else
+#elif HAVE_NEURON
 #include "nccl-headers/neuron/error.h"
```
Comment (bwbarrett) [inline include/nccl-headers/error.h:11]: Nothing sane will happen in the third case; add an else case that `#error`s.

**Review by aws-nslick — APPROVED** [review]
Note that https://github.com/aws/aws-ofi-nccl/pull/351/ has the same fix and integrates with clang-tidy to prevent this in perpetuity.

**Comment by aws-nslick** [comment]
Why did this pass distcheck given the unguarded usage of ncclInvalidUsage [here](https://github.com/rajachan/aws-ofi-nccl/blob/missing-config-stop-relying-on-getting-lucky/src/tuner/nccl_ofi_tuner.c#L24)?

**Comment by rauteric** [comment]
> Why did this pass distcheck given the unguarded usage of ncclInvalidUsage [here](https://github.com/rajachan/aws-ofi-nccl/blob/missing-config-stop-relying-on-getting-lucky/src/tuner/nccl_ofi_tuner.c#L24)?

I think it's because `nccl_ofi_tuner.c` includes `nccl-headers/nvidia/tuner.h` (which includes `nccl-headers/net.h` which includes `nccl-headers/neuron/net.h` which includes `nccl-headers/neuron/error.h`) before including `nccl_ofi_tuner.h` (which includes `config.h` which defines `HAVE_CUDA`) 🙃🙃🙃 

tl;dr `nccl_ofi_tuner.c` should include `nccl_ofi_tuner.h` first.

**Comment by bwbarrett** [comment]
> tl;dr `nccl_ofi_tuner.c` should include `nccl_ofi_tuner.h` first.

Incorrect.  The first include in any `.c` file should be `#include "config.h"`.  I missed this in CR, but that needs to be fixed.

File: include/nccl_ofi_tuner.h [hunk 2]
```diff
@@ -3,6 +3,7 @@
 
 #include <linux/limits.h>
 #include <float.h>
+#include "config.h"
```
Comment (bwbarrett) [inline include/nccl_ofi_tuner.h:6]: config.h should be the first include, before even system header files.

**Review by bwbarrett — CHANGES_REQUESTED** [review]

**Review by bwbarrett — DISMISSED** [review]

**Comment by bwbarrett** [comment]
The failure looks like a real issue:
```
In file included from ../include/nccl-headers/nvidia/tuner.h:12,
                 from tuner/nccl_ofi_model.c:3:
../include/nccl-headers/error.h:13:2: error: #error "Neither CUDA nor Neuron support is available"
   13 | #error "Neither CUDA nor Neuron support is available"
      |  ^~~~~
make[2]: *** [Makefile:559: tuner/nccl_ofi_model.lo] Error 1
make[1]: *** [Makefile:439: all-recursive] Error 1
make: *** [Makefile:371: all] Error 2
```

**Review by AmedeoSapio — APPROVED** [review]
